### PR TITLE
feat: allow synopsis moderator to show progress

### PR DESCRIPTION
**Pull request by michaelneale**
This adds status reporting to synopsis and moderators (gives better visual feedback) - does make it seem wierd to have that initial summary (and this also will only summarize after a few iterations of rewrite - to stop the initial slowstart)

### CODE REVIEW

File: src/goose/synopsis/moderator.py [hunk 1]
```diff
@@ -33,19 +35,25 @@ class Synopsis(Moderator):
...
         hints_path = Path(".goosehints")
         home_hints_path = Path.home() / ".config/goose/.goosehints"
         if hints_path.is_file():
             hints.append(render_template(hints_path))
+            if self.notifier:
+                self.notifier.status("Loaded local hints")
         if home_hints_path.is_file():
             hints.append(render_template(home_hints_path))
+            if self.notifier:
+                self.notifier.status("Loaded global hints")
```
Comment (lamchau) [inline src/goose/synopsis/moderator.py:56]: might be extra stuff on the screen but if we show the full path to hints on iTerm (and maybe other terminal emulators) it'll create a hyperlink to click when a user holds down ⌘ cmd.

File: src/goose/synopsis/moderator.py [hunk 2]
```diff
@@ -56,48 +64,70 @@ def rewrite(self, exchange: Exchange) -> None:
...
+            self.notifier.status("Summarizing context")
         return new_exchange.generate().content[0].text
 
     def plan(self, exchange: Exchange) -> str:
         message = Message.load("plan.md", synopsis=self, messages=self.originals, exchange=exchange, system=system)
         model = os.environ.get("PLANNER", exchange.model)
         new_exchange = exchange.replace(moderator=PassiveModerator(), tools=(), system="", messages=[], model=model)
         new_exchange.add(message)
+        if self.notifier:
+            self.notifier.status("Generating plan with model...")
```
Comment (lamchau) [inline src/goose/synopsis/moderator.py:132]: might just be a dev thing but i swap models often, add the model name to the output?

File: src/goose/synopsis/moderator.py (hunk 1 above)
Comment (michaelneale) [reply src/goose/synopsis/moderator.py:56]: @lamchau yeah this is only for moderating at the moment - but that is a cool idea- will add it elsewhere

File: src/goose/synopsis/moderator.py (hunk 2 above)
Comment (michaelneale) [reply src/goose/synopsis/moderator.py:132]: good idea

**Comment by michaelneale** [comment]
@lamchau bummer this can't work for reasons I suspected - the notifier is only in goose, not exchange, so creates a sort of circular dependency. Do you think its worth the lift to move notifier to exchange as it is nice to see goose thinking?

(I will do another PR with small change for optimising summarize)

**Pull request closed without merging**